ci/cudf_pandas_scripts/pandas-tests/diff.sh

```diff
@@ -8,17 +8,22 @@
 
 # Hard-coded needs to match the version deduced by rapids-upload-artifacts-dir
 MAIN_ARTIFACT=$(rapids-s3-path)cuda12_$(arch)_py310.main-results.json
-PR_ARTIFACT=$(rapids-s3-path)cuda12_$(arch)_py310.pr-results.json
-aws s3 cp $MAIN_ARTIFACT main-results.json
+PR_ARTIFACT=$(rapids-s3-path)cuda12_$(arch)_py39.pr-results.json
+
+rapids-logger "Fetching latest available results from nightly"
+aws s3api list-objects-v2 --bucket rapids-downloads --prefix "nightly/" --query "sort_by(Contents[?ends_with(Key, '.main-results.json')], &LastModified)[::-1].[Key]" --output text > s3_output.txt
```

Comment:
Maybe this logic should be moved to `gha-tools` since it relies on the paths created there.